pkg/controller/atlasproject/atlasproject_controller.go

```diff
@@ -99,125 +100,155 @@ func (r *AtlasProjectReconciler) Reconcile(context context.Context, req ctrl.Req
+		setCondition(workflowCtx, status.ProjectReadyType, result)
 		return result.ReconcileResult(), nil
 	}
 
 	var authModes authmode.AuthModes
-	if authModes, result = r.ensureX509(ctx, projectID, project); !result.IsOk() {
-		setCondition(ctx, status.ProjectReadyType, result)
+	if authModes, result = r.ensureX509(workflowCtx, projectID, project); !result.IsOk() {
+		setCondition(workflowCtx, status.ProjectReadyType, result)
 		return result.ReconcileResult(), nil
 	}
 	authModes.AddAuthMode(authmode.Scram) // add the default auth method
-	ctx.EnsureStatusOption(status.AtlasProjectAuthModesOption(authModes))
+	workflowCtx.EnsureStatusOption(status.AtlasProjectAuthModesOption(authModes))
 
 	// Updating the status with "projectReady = true" and "IPAccessListReady = false" (not as separate updates!)
-	ctx.SetConditionTrue(status.ProjectReadyType)
+	workflowCtx.SetConditionTrue(status.ProjectReadyType)
 	r.EventRecorder.Event(project, "Normal", string(status.ProjectReadyType), "")
 
-	results := r.ensureProjectResources(ctx, projectID, project, context)
+	results := r.ensureProjectResources(workflowCtx, projectID, project, ctx)
 	for i := range results {
 		if !results[i].IsOk() {
-			logIfWarning(ctx, result)
+			logIfWarning(workflowCtx, result)
 			return results[i].ReconcileResult(), nil
 		}
 	}
 
-	ctx.SetConditionTrue(status.ReadyType)
+	workflowCtx.SetConditionTrue(status.ReadyType)
 	return workflow.OK().ReconcileResult(), nil
 }
 
-func (r *AtlasProjectReconciler) ensureDeletionFinalizer(ctx *workflow.Context, atlasClient mongodbatlas.Client, projectID string, project *mdbv1.AtlasProject, context context.Context) (result workflow.Result) {
-	log := ctx.Log
+func (r *AtlasProjectReconciler) ensureDeletionFinalizer(ctx context.Context, workflowCtx *workflow.Context, atlasClient mongodbatlas.Client, project *mdbv1.AtlasProject) (result workflow.Result) {
+	log := workflowCtx.Log
 
 	if project.GetDeletionTimestamp().IsZero() {
 		if !customresource.HaveFinalizer(project, customresource.FinalizerLabel) {
 			log.Debugw("Add deletion finalizer", "name", customresource.FinalizerLabel)
-			if err := r.addDeletionFinalizer(context, project); err != nil {
-				return workflow.Terminate(workflow.Internal, err.Error())
+			if err := customresource.ManageFinalizer(ctx, r.Client, project, customresource.SetFinalizer); err != nil {
+				return workflow.Terminate(workflow.AtlasFinalizerNotSet, err.Error())
 			}
 		}
 	}
 
 	if !project.GetDeletionTimestamp().IsZero() {
 		if customresource.HaveFinalizer(project, customresource.FinalizerLabel) {
-			if customresource.ResourceShouldBeLeftInAtlas(project) {
-				log.Infof("Not removing the Atlas Project from Atlas as the '%s' annotation is set", customresource.ResourcePolicyAnnotation)
+			log.Infow("RESOURCE PROTECTED", r.ObjectDeletionProtection, customresource.IsResourceProtected(project, r.ObjectDeletionProtection))
```

Comment:
nit: Any reason for upper case info here?

Reply:
Not a special one, seemed to be the norm when PRs were open, seems now it has not been preserved.

I would suggest to clean it up afterwards, this should be a clean rebase of what was already reviewed and merged.

Reply:
This was a debug I put in the code, I'll remove it in the upcoming PRs